packages/evm/test/precompiles/05-modexp.spec.ts

```diff
@@ -51,3 +71,81 @@ describe('Precompiles: MODEXP', () => {
     assert.strictEqual(result.executionGasUsed, gas)
   })
 })
+
+function enoughGas(opts: PrecompileInput): boolean {
+  const pName = getPrecompileName('05')
+  const data = opts.data.length < 96 ? setLengthRight(opts.data, 96) : opts.data
+
+  let adjustedELen = getAdjustedExponentLength(data, opts)
+  if (adjustedELen < BIGINT_1) {
+    adjustedELen = BIGINT_1
+  }
+
+  const bLen = bytesToBigInt(data.subarray(0, 32))
+  const mLen = bytesToBigInt(data.subarray(64, 96))
+
+  let maxLen = bLen
+  if (maxLen < mLen) {
+    maxLen = mLen
+  }
+  const Gquaddivisor = opts.common.param('modexpGquaddivisorGas')
+  let gasUsed
+
+  if (!(opts.common.isActivatedEIP(2565) === true)) {
+    gasUsed = (adjustedELen * multiplicationComplexity(maxLen)) / Gquaddivisor
+  } else {
+    gasUsed = (adjustedELen * multiplicationComplexityEIP2565(maxLen)) / Gquaddivisor
+    if (gasUsed < BIGINT_200) {
+      gasUsed = BIGINT_200
+    }
+  }
+  if (!(gasLimitCheck(opts, gasUsed, pName) === true)) {
+    return false
+  }
+  return true
+}
+
+describe('Precompiles: MODEXP with EIP-7823', async () => {
+  const common = new Common({ chain: Mainnet, eips: [7823] })
+  const evm = await createEVM({
+    common,
+  })
+  const addressStr = '0000000000000000000000000000000000000005'
+  const MODEXP = getActivePrecompiles(common).get(addressStr)!
+
+  for (let i = 1020; i < 1028; i++) {
+    const bLen = i
+    const eLen = i
+    const mLen = i
+    const maxInputLen = Math.max(bLen, eLen, mLen)
+    const base = randomBytes(bLen)
+    const exponent = randomBytes(eLen)
+    const modulo = randomBytes(mLen)
+    const expected = expMod(bytesToBigInt(base), bytesToBigInt(exponent), bytesToBigInt(modulo))
+    const input = concatBytes(
+      setLengthLeft(toBytes(bLen), 32),
+      setLengthLeft(toBytes(eLen), 32),
+      setLengthLeft(toBytes(mLen), 32),
+      concatBytes(base, exponent, modulo),
+    )
+
+    it(`MODEXP should reject and consume all gas for inputs over 8192 bits in length - case ${maxInputLen} bytes`, async () => {
+      const opts = {
+        data: input,
+        gasLimit: BigInt(0xffffffff),
+        common,
+        _EVM: evm,
+      }
+
+      // sanity check to make sure there is enough gas in order to isolate error to cases causing OOG due to size limit of inputs
+      while (!enoughGas(opts)) opts.gasLimit *= BigInt(256)
```

Comment:
If we then have the "calculate gas" method, we can thus "just set" this as the gasLimit. Even if the gas limit is enough, then it will still run out-of-gas because of the inputs being too large.